### PR TITLE
Fix timeout children (bugfix)

### PR DESCRIPTION
**Pull request by Hook25**
## Description

When a process spawns childrens and it is killed, if the childrens were non-deamons they are re-parented by the super reaper to the closest relative, falling back on init if none are found. This behaviour is not what we want for our timeout decorator, namely, when a function times out, all processes spawned by the decorated functions must be killed. This updates the decorator so that this happens.

## Resolved issues

Fixes: https://warthogs.atlassian.net/browse/CHECKBOX-1250

## Documentation

Updated docstring with the new limitations and features

## Tests

All tests are still passing

### CODE REVIEW

File: checkbox-support/checkbox_support/helpers/timeout.py [hunk 1]
```diff
@@ -71,3 +75,14 @@ def _f(*args, **kwargs):
         return _f
 
     return timeout_timeout_s
+
+
+@timeout(1)
+def main():
+    import subprocess
+
+    subprocess.check_call(["sleep", "10000"])
```
Comment (kissiel) [inline checkbox-support/checkbox_support/helpers/timeout.py:84]: leftovers?

**Review by kissiel — CHANGES_REQUESTED** [review]
some leftovers left over :)

File: checkbox-support/checkbox_support/helpers/timeout.py (hunk 1 above)
Comment (Hook25) [reply checkbox-support/checkbox_support/helpers/timeout.py]: fixed

**Review by kissiel — APPROVED** [review]
Nice battle.
Happy to see us conquering this problem.

**Pull request merged**